source/simulator/assemblers/newton_stokes.cc

```diff
@@ -630,6 +632,424 @@ namespace aspect
     }
 
 
+
+    template <int dim>
+    void
+    NewtonStokesSimplePreconditioner<dim>::
+    execute (internal::Assembly::Scratch::ScratchBase<dim>   &scratch_base,
+             internal::Assembly::CopyData::CopyDataBase<dim> &data_base) const
+    {
+      internal::Assembly::Scratch::StokesPreconditioner<dim> &scratch = dynamic_cast<internal::Assembly::Scratch::StokesPreconditioner<dim>&> (scratch_base);
+      internal::Assembly::CopyData::StokesPreconditioner<dim> &data = dynamic_cast<internal::Assembly::CopyData::StokesPreconditioner<dim>&> (data_base);
+
+      const Introspection<dim> &introspection = this->introspection();
+      const FiniteElement<dim> &fe = this->get_fe();
+      const unsigned int stokes_dofs_per_cell = data.local_dof_indices.size();
+      const unsigned int n_q_points           = scratch.finite_element_values.n_quadrature_points;
+      const double derivative_scaling_factor = this->get_newton_handler().parameters.newton_derivative_scaling_factor;
+      const double pressure_scaling = this->get_pressure_scaling();
+
+      const double eta = scratch.material_model_outputs.viscosities[0];
+      const double one_over_eta = 1. / eta;
+
+      AssertThrow(dynamic_cast<const MaterialModel::ViscoPlasticSimple<dim> *>(&this->get_material_model()), ExcInternalError());
+
+      // If elasticity is enabled, then we need ElasticOutputs to get the viscoelastic
+      // strain rate.
+      const MaterialModel::ElasticOutputs<dim> *elastic_out =
+        scratch.material_model_outputs.template get_additional_output<MaterialModel::ElasticOutputs<dim>>();
```

Comment:
This interface was changed in #6342. You might have to rebase to main to get that change.
```suggestion
      // If elasticity is enabled, then we need ElasticOutputs to get the viscoelastic
      // strain rate.
      const std::shared_ptr<MaterialModel::ElasticOutputs<dim>> elastic_out =
        scratch.material_model_outputs.template get_additional_output_object<MaterialModel::ElasticOutputs<dim>>();
```